.github/workflows/tests.yml

```
@@ -132,9 +132,7 @@ jobs:
       - name: Get pyright version
         id: pyright_version
         run: |
-          PYRIGHT_VERSION=$(grep pyright== requirements-tests.txt | cut -d "#" -f 1 | cut -d \; -f 1 | cut -d = -f 3)
-          echo pyright version: "${PYRIGHT_VERSION}"
-          echo PYRIGHT_VERSION="${PYRIGHT_VERSION}" >> "${GITHUB_OUTPUT}"
+          echo PYRIGHT_VERSION=$(uv pip freeze | grep -e ^pyright== | cut -d = -f 3) | tee $GITHUB_OUTPUT
```

So that it looks at the global environment rather than the activated virtualenv that doesn't have pyright-python installed into it:

```suggestion
          echo PYRIGHT_VERSION=$(uv pip freeze --system | grep -e ^pyright== | cut -d = -f 3) | tee $GITHUB_OUTPUT
```

You might also need to move this step before the "create an isolated environment for testing" and "activate the environment for the rest of the job" steps. We install pyright-python globally via the `uv pip install -r requirements-tests.txt --system` step, but we don't install it into the isolated environment.